plugin/pkg/managedseed/validator/admission.go

```diff
@@ -0,0 +1,463 @@
+		gardenletConfig, err := helper.ConvertGardenletConfigExternal(gardenlet.Config)
+		if err != nil {
+			return append(allErrs, field.Invalid(configPath, gardenlet.Config, fmt.Sprintf("could not decode config: %v", err))), nil
+		}
+
+		if gardenletConfig.SeedConfig != nil {
+			seedConfigPath := fldPath.Child("seedConfig")
+
+			// Convert seed config to internal version
+			seed, err := helper.ConvertSeed(&gardenletConfig.SeedConfig.Seed)
+			if err != nil {
+				return allErrs, apierrors.NewInternalError(fmt.Errorf("could not convert seed config: %v", err))
+			}
+
+			// Admit seed spec against shoot
+			errs, err := v.admitSeedSpec(ctx, &seed.Spec, shoot, seedConfigPath.Child("spec"))
+			if err != nil {
+				return allErrs, err
+			}
+			allErrs = append(allErrs, errs...)
+
+			// Set seed config back to gardenletConfig.SeedConfig.Seed
+			seedx, err := helper.ConvertSeedExternal(seed)
+			if err != nil {
+				return allErrs, apierrors.NewInternalError(fmt.Errorf("could not convert seed config: %v", err))
+			}
+			gardenletConfig.SeedConfig.Seed = *seedx
+		}
+
+		// Set gardenlet config back to gardenlet.Config
+		gardenlet.Config = gardenletConfig
+	}
+
+	return allErrs, nil
+}
+
+func (v *ValidateManagedSeed) admitSeedSpec(ctx context.Context, spec *gardencore.SeedSpec, shoot *gardencore.Shoot, fldPath *field.Path) (field.ErrorList, error) {
+	var allErrs field.ErrorList
+
+	// Initialize backup provider
+	if spec.Backup != nil && spec.Backup.Provider == "" {
+		spec.Backup.Provider = shoot.Spec.Provider.Type
+	}
+
+	// Initialize and validate DNS and ingress
+	ingressDomain := fmt.Sprintf("%s.%s", common.IngressPrefix, *(shoot.Spec.DNS.Domain))
+	if spec.Ingress != nil {
+		if gardencorehelper.NginxIngressEnabled(shoot.Spec.Addons) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("ingress"), fmt.Sprintf("seed ingress controller cannot be enabled on shoot %s", kutil.ObjectName(shoot))))
+		}
+
+		if spec.DNS.Provider == nil {
+			dnsProvider, err := v.getSeedDNSProvider(ctx, shoot)
+			if err != nil {
+				if apierrors.IsInternalError(err) {
+					return allErrs, err
+				}
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("ingress"), spec.Ingress, err.Error()))
+			}
+			spec.DNS.Provider = dnsProvider
+		}
+		// TODO If spec.DNS.Provider is not nil should we check if the user specified exactly the DNS provider we found above?
+		// I assume not, since the shoot may have multiple DNS providers
+
+		if spec.Ingress.Domain == "" {
+			spec.Ingress.Domain = ingressDomain
+		} else if spec.Ingress.Domain != ingressDomain {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ingress", "domain"), spec.Ingress.Domain, fmt.Sprintf("seed ingress domain must be equal to shoot DNS domain %s", ingressDomain)))
+		}
+	} else {
+		if spec.DNS.IngressDomain == nil || *spec.DNS.IngressDomain == "" {
+			spec.DNS.IngressDomain = &ingressDomain
```

Comment:
This code was basically copied from the existing `use-as-seed` implementation. I had some doubts but decided to keep it as it is, I simply wasn't sure. 
Should I then check if the ingress addon is enabled, and only if it is, set the ingress domain, and otherwise leave it empty?

Reply:
Based on the above, I added a check for the ingress addon being enabled.

Reply:
I think, it's good now like you did it.